scripts/msg.h.template

```diff
@@ -173,6 +173,20 @@ ros::message_operations::Printer< @(cpp_full_name_with_alloc) >::stream(s, "", v
 return s;
 }
 
+@# Equality
+template<typename ContainerAllocator>
+bool operator==(const @(cpp_full_name_with_alloc) & lhs, const @(cpp_full_name_with_alloc) & rhs)
```

Comment:
Maybe allow comparison between objects using different allocator types?

Something like
```c++
template<typename ContainerAllocator1, typename ContainerAllocator2>
bool operator==(const MsgType<ContainerAllocator1>& lhs, const MsgType<ContainerAllocator2>& rhs)
{}
```

Reply:
@DangitBen can you please respond to this question.

Reply:
Sorry! I guess I lost the notification for @stertingen's question. I think I can add this tomorrow.

Out of curiosity, why is this something that you wanted?